docs/TSCONFIG.md

```diff
@@ -0,0 +1,150 @@
+# TsConfig files structure
+
+The presence of a config file in a directory indicates that it is the root of a TypeScript project.
+In Stark, several types of tsconfig files are placed in the packages for different purposes, described below.
+
+## tsconfig.json
+
+The `tsconfig.json` file can be found at the following locations:
+
+```txt
++---starter
+|	tsconfig.json
++---showcase
+|	tsconfig.json
++---packages
+	tsconfig.json
+	+---stark-build
+	|	tsconfig.json
+...
+```
+
+The structure of this file is like the following:
+
+```
+{
+	...
+  "extends": "./node_modules/@nationalbankbelgium/stark-build/tsconfig.json",
+  "compilerOptions": {
+    "baseUrl": "./src",
+    "outDir": "./dist",
+    ...
+    }
+  },
+  "exclude": ["node_modules", "dist"]
+  ...
+}
+```
+
+This file is used simply to build the packages where it is located.
+
+## tsconfig.app.json
+
+The `tsconfig.app.json` file is located in the following folders:
+
+```txt
++---starter
+|	tsconfig.app.json
++---showcase
+|	tsconfig.app.json
+...
+```
+
+Those files are used for the build the application itself.
+
+## tsconfig.spec.json
+
+The `tsconfig.spec.json` file can be found at the following locations:
+
+```txt
++---starter
+|	tsconfig.spec.json
++---showcase
+|	tsconfig.spec.json
++---packages
+	+---stark-core
+	|	tsconfig.spec.json
+	+---stark-ui
+	|	tsconfig.spec.json
+...
+```
+
+The structure of this files is as the following:
+
+```
+"extends": "./tsconfig-build.json",
+  "compilerOptions": {
+    "module": "commonjs",
+    "paths": {
+      "@nationalbankbelgium/stark-core/testing": ["../../dist/packages/stark-core/testing"],
+      ...
+    }
+  },
+  "files": null,
+  "include": ["**/*.ts"]
+}`
+```
+
+Those files are used only to build testing packages.
```

Comment:
It is not really the testing packages. These files are used to build the testing bundles which contain the spec.ts files and all the files under test.

On top of that, the code is compiled using CommonJs modules since we don't use Webpack to create the bundles but Karma-Typescript.